services/src/main/java/com/chatbot/services/pubsubservices/PubSubPullSubscriber.java

```diff
@@ -0,0 +1,134 @@
+import com.chatbot.services.protobuf.TriggerEventNotificationOuterClass.TriggerEventNotification.Event;
+import com.google.api.gax.batching.FlowControlSettings;
+import com.google.api.gax.core.ExecutorProvider;
+import com.google.api.gax.core.InstantiatingExecutorProvider;
+import com.google.cloud.pubsub.v1.AckReplyConsumer;
+import com.google.cloud.pubsub.v1.MessageReceiver;
+import com.google.cloud.pubsub.v1.Subscriber;
+import com.google.protobuf.Struct;
+import com.google.pubsub.v1.ProjectSubscriptionName;
+import com.google.pubsub.v1.PubsubMessage;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+
+// Handle messages from cloud sub using a pull subscriber
+// Annotating this class with @Component will launch the pull subscriber on application start up
+public class PubSubPullSubscriber {
+
+  private final HangoutsAsyncService asyncService;
+  private static final Logger logger = LoggerFactory.getLogger(PubSubPullSubscriber.class);
+  private static String projectID;
+  private static String subscriptionID;
+  private static Long maxOutstandingElements;
+  private static Long maxOutstandingBytes;
+  private static int threadCount;
+
+  PubSubPullSubscriber(
+      @Value("${pubsubConfig.maxOutstandingElements}") final String maxOutstandingElementsToSet,
+      @Value("${pubsubConfig.maxOutstandingBytes}") final String maxOutstandingBytesToSet,
+      @Value("${pubsubConfig.threadCount}") final String threadCountToSet,
+      final HangoutsAsyncService asyncServiceToSet) throws InterruptedException {
+    asyncService = asyncServiceToSet;
+    projectID = System.getenv("projectID");
+    subscriptionID = System.getenv("subscriptionID");
+    maxOutstandingElements = Long.parseLong(maxOutstandingElementsToSet);
+    maxOutstandingBytes = Long.parseLong(maxOutstandingBytesToSet);
+    threadCount = Integer.parseInt(threadCountToSet);
+    launchSubscriber();
+  }
+
+  // launch the pull subscribers
+  private void launchSubscriber()
+      throws InterruptedException, IllegalArgumentException {
+    final ProjectSubscriptionName subscriptionName = ProjectSubscriptionName
+        .of(projectID, subscriptionID);
+    final MessageReceiver receiver =
+        (final PubsubMessage message, final AckReplyConsumer consumer) -> {
+      final String messageData = message.getData().toStringUtf8();
+      final Map<String, String> messageAttributesMap = message.getAttributesMap();
+      if(messageData.equals(ChatServiceConstants.TRIGGER_EVENT_MESSAGE)) {
+        final TriggerEventNotification triggerEventNotification = 
+            buildNotificationFromMessage(messageAttributesMap);
+        try {
+          asyncService.triggerEventHandler(triggerEventNotification);
+        } catch (final Exception e) {
+          logger.error("Error while handling event trigger", e);
+        }
```

Comment:
Don't use failsafe, throw exception

Reply:
This is in a lambda so need to use try-catch

Reply:
The statement just below this one is throwing exception and it belongs inside lambda, how is that working?
Look for "handling checked exceptions in lambda"

Reply:
The other one works because IllegalArgumentException is an unchecked exception and IOException is a Checked exception. Can't seem to find a direct way to throw checked exception from a lambda.

Reply:
Add a TODO there to remove the catch now

Reply:
added